app/src/main/java/io/gnosis/safe/ui/assets/coins/CoinsFragment.kt

```diff
@@ -49,15 +50,24 @@ class CoinsFragment : BaseViewBindingFragment<FragmentCoinsBinding>() {
                     binding.refresh.isRefreshing = state.refreshing
                     state.viewAction?.let { action ->
                         when (action) {
-                            is UpdateBalances -> adapter.setItems(action.newBalances)
+                            is UpdateBalances -> {
+                                binding.contentNoData.root.visibility = View.GONE
+                                binding.refresh.visible(true)
+
+                                adapter.setItems(action.newBalances)
+                            }
                             is ShowError -> {
                                 hideLoading()
+
+                                binding.contentNoData.root.visible(true)
```

Comment:
here we have to check if we already have some data and show placeholder only if nothing was previously loaded for this safe